src/Entities/Pawn.cpp

```diff
@@ -423,18 +423,26 @@ void cPawn::HandleFalling(void)
 				Damage = static_cast<int>(static_cast<float>(Damage) * 0.33);
 			}
 
+			const auto Below = POS_TOINT.addedY(-1);
+			const auto BlockBelow = GetWorld()->GetBlock(Below);
+
+			if (BlockBelow == E_BLOCK_HAY_BALE)
+			{
+				Damage = std::clamp(static_cast<int>(static_cast<float>(Damage) * 0.2), 1, 20);
+			}
+
```

Comment:
This should be moved inside the `if (Below.y >= 0)` block, to guard the GetBlock.

Reply:
> 
> 
> This should be moved inside the `if (Below.y >= 0)` block, to guard the GetBlock.

Should I move TakeDamage in it or move it after if? By practice seems like there is no damage to player applied if there is no particles, but I believe that there was a reason to do as it is now.

Reply:
I would move the TakeDamage to after the if, so the update to Damage by the hay bale check is applied. I don't _think_ that it would do any harm to put it inside the if, but keeping the behaviour the same as it is now (by moving it after) just makes sure that we don't introduce any new bugs.